### PR TITLE
Fix subgraph issues

### PR DESCRIPTION
**Pull request by solidoracle**

### CODE REVIEW

**Review by bengobeil — APPROVED** [review]
looks good, didn't know these changes were causing issues

File: bulla-contracts/schema.graphql [hunk 1]
```diff
@@ -265,6 +265,21 @@ type DepositMadeEvent implements IEventLog & IPoolTransaction @entity {
   ipfsHash: String # nullable attachment property
 }
 
+type DepositMadeWithAttachmentEvent implements IEventLog & IPoolTransaction @entity {
```
Comment (bengobeil) [inline bulla-contracts/schema.graphql:268]: not sure why we are adding this back in

File: bulla-contracts/src/mappings/BullaFactoring.ts [hunk 2]
```diff
@@ -284,11 +286,42 @@ export function handleDepositV2(event: Deposit): void {
   handleDeposit(event, "v2");
 }
 
-export function handleDepositMadeWithAttachment(event: DepositMadeWithAttachment): void {
+// @notice V1 does not have ERC4626 functions and events, so cannot reuse logic from V2
+export function handleDepositMadeWithAttachmentV1(event: DepositMadeWithAttachment): void {
   const ev = event.params;
+  const DepositMadeWithAttachmentEvent = createDepositMadeWithAttachmentEvent(event);
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaFactoring.ts:292]: to conform with the V2 design, we would create a normal DepositMade event and assign the ipfsHash to it.

**Review by bengobeil — CHANGES_REQUESTED** [review]

File: bulla-contracts/src/mappings/BullaFactoring.ts (hunk 2 above)
Comment (solidoracle) [reply bulla-contracts/src/mappings/BullaFactoring.ts:292]: `Argument of type 'DepositMadeWithAttachment' is not assignable to parameter of type 'Deposit'.`

Comment (solidoracle) [reply bulla-contracts/src/mappings/BullaFactoring.ts:292]: that's why I added back `DepositMadeWithAttachmentEvent` in the schema

File: bulla-contracts/src/functions/BullaFactoring.ts [hunk 3]
```diff
@@ -49,12 +50,16 @@ export const createInvoiceUnfactoredEventv1 = (underlyingTokenId: BigInt, event:
...
-  return "DepositMade-" + poolAddress.toHexString() + "-" + event.transaction.hash.toHexString() + "-" + event.logIndex.toString();
+  return (
+    "DepositMade-" + poolAddress.toHexString() + "-" + event.transaction.hash.toHexString() + "-" + (logIndexOverride ? logIndexOverride : event.logIndex).toString()
+  );
 };
 
-export const createDepositMadeEvent = (event: Deposit): DepositMadeEvent => new DepositMadeEvent(getDepositMadeEventId(event));
+export const createDepositMadeEventV2 = (event: Deposit): DepositMadeEvent => new DepositMadeEvent(getDepositMadeEventId(event, null));
+
+export const createDepositMadeEventV1 = (event: DepositMadeWithAttachment): DepositMadeEvent => new DepositMadeEvent(getDepositMadeEventId(event, null));
```
Comment (bengobeil) [inline bulla-contracts/src/functions/BullaFactoring.ts:62]: we still need to create the DepositMadeEvent from the DepositMade event in V1

File: bulla-contracts/src/mappings/BullaFactoring.ts [hunk 4]
```diff
@@ -248,7 +250,7 @@ export function handleInvoiceUnfactoredV2(event: InvoiceUnfactored): void {
 export function handleDeposit(event: Deposit, version: string): void {
   const ev = event.params;
 
-  const DepositMadeEvent = createDepositMadeEvent(event);
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaFactoring.ts:251]: Works for V2 but not for V1. V1 does not have `Deposit` event, rather `DepositMade` event

**Review by bengobeil — CHANGES_REQUESTED** [review]

File: bulla-contracts/src/mappings/BullaFactoring.ts [hunk 5]
```diff
@@ -278,18 +279,41 @@ export function handleDeposit(event: Deposit, version: string): void {
...
+  historical_factoring_statistics.save();
 }
 
 export function handleDepositMadeWithAttachmentV1(event: DepositMadeWithAttachment): void {
   const ev = event.params;
 
-  const DepositMadeEvent = createDepositMadeEventV1(event);
+  const DepositMadeEvent = createDepositMadeWithAttachmentEventV1(event);
 
   DepositMadeEvent.poolAddress = event.address;
```
Comment (bengobeil) [inline bulla-contracts/src/mappings/BullaFactoring.ts:318]: ipfsHash is not being set in this function

File: bulla-contracts/template.yaml [hunk 6]
```diff
@@ -260,8 +260,8 @@ dataSources:
           handler: handleInvoicePaidV1
         - event: InvoiceUnfactored(indexed uint256,address,uint256,uint256)
           handler: handleInvoiceUnfactoredV1
-        - event: Deposit(indexed address,indexed address,uint256,uint256)
-          handler: handleDepositV1
+        - event: DepositMade(indexed address,uint256,uint256)
+          handler: handleDepositMadeV1
         - event: DepositMadeWithAttachment(indexed address,uint256,uint256,(bytes32,uint8,uint8))
           handler: handleDepositMadeWithAttachment
```
Comment (bengobeil) [inline bulla-contracts/template.yaml:266]: handleDepositMadeWithAttachmentV1

**Review by bengobeil — APPROVED** [review]

**Review by bengobeil — APPROVED** [review]

**Pull request merged**